components/browser/search/src/test/java/mozilla/components/browser/search/provider/AssetsSearchEngineProviderTest.kt

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.search.provider
+
+import kotlinx.coroutines.experimental.runBlocking
+import mozilla.components.browser.search.SearchEngine
+import mozilla.components.browser.search.provider.filter.SearchEngineFilter
+import mozilla.components.browser.search.provider.localization.SearchLocalizationProvider
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class AssetsSearchEngineProviderTest {
+    @Test
+    fun `Load search engines for en-US from assets`() = runBlocking {
+        val localizationProvider = object : SearchLocalizationProvider() {
+            override val country: String = "US"
+            override val language = "en"
+            override val region: String? = null
+        }
+
+        val searchEngineProvider = AssetsSearchEngineProvider(localizationProvider)
+
+        val searchEngines = searchEngineProvider.loadSearchEngines(RuntimeEnvironment.application)
+
+        assertEquals(6, searchEngines.size)
+    }
+
+    @Test
+    fun `Load search engines for en-US with with filter`() = runBlocking {
+        val localizationProvider = object : SearchLocalizationProvider() {
+            override val country: String = "US"
+            override val language = "en"
+            override val region: String? = null
+        }
+
+        val filter = object : SearchEngineFilter {
+            private val exclude = listOf("yahoo", "bing", "ddg")
+
+            override fun filter(searchEngine: SearchEngine): Boolean {
+                return !exclude.contains(searchEngine.identifier)
+            }
+        }
+
+        val searchEngineProvider = AssetsSearchEngineProvider(localizationProvider, listOf(filter))
+
+        val searchEngines = searchEngineProvider.loadSearchEngines(RuntimeEnvironment.application)
+
+        assertEquals(4, searchEngines.size)
+    }
+
+    @Test
+    fun `Load search engines for de-DE with US region override`() = runBlocking {
+        run {
+            val localizationProviderWithoutRegion = object : SearchLocalizationProvider() {
+                override val country: String = "DE"
+                override val language = "de"
+                override val region: String? = null
+            }
+
+            val searchEngineProvider = AssetsSearchEngineProvider(localizationProviderWithoutRegion)
+            val searchEngines = searchEngineProvider.loadSearchEngines(RuntimeEnvironment.application)
+
+            assertEquals(7, searchEngines.size)
```

Comment:
\[detekt\] INFO: \[style/MagicNumber\] Report magic numbers\. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is\. It's better to declare such numbers as constants and give them a proper name\. By default, \-1, 0, 1, and 2 are not considered to be magic numbers\.